src/providers/gif.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{
+    render::{display::ContentProvider, scheduler::ContentWrapper, gif},
+    scheduler::CONTENT_PROVIDERS,
+};
+use anyhow::Result;
+use apex_hardware::FrameBuffer;
+use async_stream::try_stream;
+use config::Config;
+use embedded_graphics::{
+    geometry::Point,
+};
+use futures::Stream;
+use linkme::distributed_slice;
+use log::info;
+use tokio::{
+    time,
+    time::{Duration, MissedTickBehavior},
+};
+use std::fs::File;
+
+
+
+
+#[doc(hidden)]
+#[distributed_slice(CONTENT_PROVIDERS)]
+pub static PROVIDER_INIT: fn(&Config) -> Result<Box<dyn ContentWrapper>> = register_callback;
+
+#[doc(hidden)]
+#[allow(clippy::unnecessary_wraps)]
+fn register_callback(config: &Config) -> Result<Box<dyn ContentWrapper>> {
+    info!("Registering Gif display source.");
+
+    let gif_path = config.get_str("gif.path").unwrap();
+    let gif_file = File::open(&gif_path);
+
+    let gif = match gif_file {
+        Ok(file) => gif::Gif::new(Point::new(0, 0), Point::new(128, 40), file),
+        Err(err) => {
+            log::error!("Failed to open GIF file '{}': {}", gif_path, err);
+			
+            // Use the `new_error` function to create an error GIF
+            let error_gif = gif::Gif::new_error(Point::new(0, 0), Point::new(128, 40));
+            error_gif

```

Comment:
Can be simplified to just `gif::Gif::new_error(Point::new(0, 0), Point::new(128, 40))`

Reply:
You don't need to even create a variable here but that's more of a nitpick